op-challenger/game/fault/contracts/faultdisputegame.go

```diff
@@ -232,8 +232,8 @@ func (f *FaultDisputeGameContract) addLocalDataTx(data *types.PreimageOracleData
 	call := f.contract.Call(
 		methodAddLocalData,
 		data.GetIdent(),
-		new(big.Int).SetBytes(data.LocalContext.Bytes()),
-		new(big.Int).SetUint64(uint64(data.OracleOffset)),
+		data.GetLocalContextBigInt(),
+		data.GetOracleOffsetBigInt(),
```

Comment:
hmm, I'm not sure this is the right approach.  This conversion is tested in `faultdisputegame_test` `TestUpdateOracleTx`.  We want to keep the conversion between the business logic type (`PreimageOracleData`) and the contract call type inside `FaultDisputeGameContract` since it's primary job is to be the translation layer between the challenger and contracts.

In terms of testing we can check the conversion in either place but both the old and new tests have the issue where the test code and production code are both using the same logic to calculate the value.  The contract tests are slightly better because they also check that the type matches the ABI, but still not great that the test and production code are just a copy of each other.

Long term I think the answer here is to change the contract type to a `bytes32` since it's a hash not a number and then the conversion goes away, but in the mean time it probably makes sense to hard code the expected values in the tests so they're at least manually verified.

Reply:
> This conversion is tested in `faultdisputegame_test` `TestUpdateOracleTx`

This is not an explicit test though

Reply:
Agree with the long term approach of using a hash

Reply:
I'm not sure what you mean by an explicit test.  The `FaultDisputeGameContract` struct's one job is to translate from the challenger domain to the contract domain of which this conversion is a part.  And the tests are actually really very effective at confirming that translation is done accurately.

We could define the "unit" we test to be even more fine-grained to have a test just for converting a hash to a number and if it were a public utility function then we should do that. But I'm not seeing a reason that the details should leak outside `FaultDisputeGameContract`. Everywhere else in challenger should always treat it as a hash because the `FaultDisputeGameContract` is providing an anti-corruption layer to ensure that the contract's view of this value doesn't leak into the challenger. Nothing else should ever be aware that this translation is happening.

I'm also having a go at pretending to be a solidity dev and changing the type in the contracts. :)